hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -213,6 +224,27 @@ protected void commit(HoodieTable table, String commitActionType, String instant
                       List<HoodieWriteStat> stats) throws IOException {
     LOG.info("Committing " + instantTime + " action " + commitActionType);
     HoodieActiveTimeline activeTimeline = table.getActiveTimeline();
+    // do save internal schema to support Implicitly add columns in write process
```

Comment:
agree

Reply:
> We need to introduce a config to enable/disable schema evolution. Only if the config is enables, we should let any side-effects of schema evolution to take effect.

may no need introduce a config.  only when user execute alter SQL /alter api explicitly, schema evolution will take effect； Otherwise, everything goes through the original process